devito/types/dense.py

```diff
@@ -243,7 +244,20 @@ def shape_with_halo(self):
         """
         return tuple(j + i + k for i, (j, k) in zip(self.shape, self._size_outhalo))
 
-    _shape_with_outhalo = shape_with_halo
+    @cached_property
+    def _shape_with_outhalo(self):
```

Comment:
We gotta find a better way, duplication like that are not good.

Reply:
Yeah, need to figure that one out. Wanted to get it working first though

Reply:
Now improved somewhat

Reply:
Well no, the docstring is not really the issue, the content of the function is which is twice the same code.